app/src/main/java/com/github/mateo762/myapplication/room/RoomModule.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.mateo762.myapplication.room
+
+import android.content.Context
+import androidx.room.Room
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RoomModule {
+    @Provides
+    @Singleton
+    fun provideDatabase(@ApplicationContext context: Context): ApplicationDatabase {
+        return Room.databaseBuilder(
+            context,
+            ApplicationDatabase::class.java,
+            "procrastinot.db"
```

Comment:
Maybe "procrastinot.db" should be stored in a constant to avoid hardcoding it.

Reply:
In testing there is an alternative database generated to test the Hilt module which is differentiated by name from the production one, so keeping it easily configurable seemed the most straightforward option. Will still keep it in mind for the future though 👍